src/components/form/fields/number/Number.tsx

```diff
@@ -0,0 +1,51 @@
+import { ChangeEvent, ChangeEventHandler } from 'react';
+import { ReqoreInput } from '@qoretechnologies/reqore';
+import { IReqoreInputProps } from '@qoretechnologies/reqore/dist/components/Input';
+
+export interface INumberFormFieldProps
+  extends Omit<IReqoreInputProps, 'value' | 'onChange' | 'type'> {
+  value: number;
+  onChange?(value: number): void;
+  type?: 'int' | 'float';
+}
+
+export const FormNumberField = ({
+  onChange,
+  autoFocus,
+  type = 'int',
+  ...rest
+}: INumberFormFieldProps) => {
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (
+    event: ChangeEvent<HTMLInputElement>
+  ): void => {
+    onChange?.(type === 'int' ? parseInt(event.target.value) : parseFloat(event.target.value));
+  };
+
+  const handleResetClick = (): void => {
+    onChange(0);
```

Comment:
`onChange(undefined)` because 0 can be a valid value

Reply:
nice catch!